### PR TITLE
winhttp callback handle error case

### PR DESCRIPTION
**Pull request by tracygooo**
- Invoke callback with error code when sending HTTP request fails
- Correct some typos

### CODE REVIEW

File: include/boost/winasio/winhttp/winhttp_asio.hpp [hunk 1]
```diff
@@ -55,6 +55,9 @@ template <typename Executor> class asio_request_context {
   // need to invoke query data inside
   net::windows::overlapped_ptr on_read_complete;
 
+  // need to invoke send http request failure
+  net::windows::overlapped_ptr on_send_request_failed;
```
Comment (youyuanwu) [inline include/boost/winasio/winhttp/winhttp_asio.hpp:59]: What function will this invoke? where is this initialized. I think currently it is empty.

File: include/boost/winasio/winhttp/winhttp_asio.hpp [hunk 2]
```diff
@@ -135,6 +138,7 @@ void __stdcall BasicAsioAsyncCallback(HINTERNET hInternet, DWORD_PTR dwContext,
     BOOST_LOG_TRIVIAL(debug) << "winhttp callback error: " << err;
     ec.assign(pAR->dwError, boost::asio::error::get_system_category());
     // TODO: invoke handler
+    cpContext->on_send_request_failed.complete(ec, 0);
```
Comment (youyuanwu) [inline include/boost/winasio/winhttp/winhttp_asio.hpp:141]: you need to switch on ctx_state_type and invoke the complete() respectively.

File: include/boost/winasio/winhttp/winhttp_asio.hpp (hunk 1 above)
Comment (tracygooo) [reply include/boost/winasio/winhttp/winhttp_asio.hpp:59]: I thought the complete function of the overlapped pointer is enough to finish the callback. Do we need a separate callback function for error handling? I saw all other cases just used complete().

File: include/boost/winasio/winhttp/winhttp_asio.hpp (hunk 2 above)
Comment (tracygooo) [reply include/boost/winasio/winhttp/winhttp_asio.hpp:141]: Will add 👍

File: include/boost/winasio/winhttp/winhttp_asio.hpp (hunk 1 above)
Comment (youyuanwu) [reply include/boost/winasio/winhttp/winhttp_asio.hpp:59]: You added an empty completion ptr, where it is no-op. So your code does not do anything.

**Comment by youyuanwu** [comment]
Superseded by #19

**Pull request closed without merging**